R/dev_utils.R

```diff
@@ -525,3 +525,105 @@ valid_time_units <- function() {
 contains_vars <- function(arg) {
   inherits(arg, "quosures") && all(map_lgl(arg, quo_is_symbol) | names(arg) != "")
 }
+
+#' Get a New Temporary Variable Name for a Dataset
+#'
+#' @param dataset The input dataset
+#' @param prefix The prefix of the new temporary variable name to create
+#'
+#' @details
+#' The function returns a new unique temporary variable name to be used inside
+#' `dataset`. The temporary variable names have the structure `prefix_n` where
+#' `n` is an integer, e.g. `tmp_var_1`. If there is alredy a variable inside
+#' `datset` with a given `prefix` then the suffix is increased by 1, e.g. if
+#' `tmp_var_1` already exists then `get_new_tmp_var()` will return `tmp_var_2`.
+#'
+#' @seealso [remove_tmp_vars()]
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' data(admiral_adsl)
+#' adsl <- admiral_adsl
+#' tmp_var <- get_new_tmp_var(adsl)
+#' mutate(adsl, !!tmp_var := NA)
+get_new_tmp_var <- function(dataset, prefix = "tmp_var") {
+  assert_data_frame(dataset, optional = TRUE)
+  assert_character_scalar(prefix)
+  if (!str_detect(prefix, "^tmp_")) {
+    abort("`prefix` must start with 'tmp_'")
+  }
+
```

Comment:
Is the prefix `tmp` or `tmp_var`?  if they do `tmp_vvv` then it won't abort?

Should we just not let developers create their own prefixes since it will abort when they put something else in there?  If we want to keep, can we have a unit test for this?

Reply:
I think the idea is that temporary variables should always start with "tmp_". Then they can be easily identified when browsing the dataset. To make the names more meaningful other names than "tmp_var_" can be specified, e.g., "tmp_date_" or "tmp_obs_nr_".

@thomas-neitmann , please correct me if I am wrong.

I have added a unit test for the error.